tests/api2/test_440_snmp.py

```diff
@@ -63,65 +132,254 @@ def validate_snmp_get_sysname_uses_same_ip(hostip):
+def test_20_validate_SNMPv3_authPriv_user_add(request):
+    """
+    Confirm we can add an SNMPv3 authPriv user
+    """
+    depends(request, ["SNMP_STARTED"], scope="module")
+    # Reset the systemd restart counter
+    reset_systemd_svcs("snmpd snmp-agent")
+
+    call('snmp.update', SNMP_USER_CONFIG)
+    assert get_systemctl_status('snmp-agent') == "RUNNING"
+
+    res = call('snmp.get_snmp_users')
+    assert SNMP_USER_NAME in res
+
+
+def test_25_validate_SNMPv3_user_function(request):
+    depends(request, ["SNMPv3_USER_ADD"], scope="module")
+    res = user_list_users(SNMP_USER_CONFIG)
+    assert SNMP_USER_NAME in res
+
+
+def test_30_validate_SNMPv3_user_retained_across_service_restart(request):
+    depends(request, ["SNMPv3_USER_ADD"], scope="module")
+    # Reset the systemd restart counter
+    reset_systemd_svcs("snmpd snmp-agent")
+
+    res = call('service.stop', 'snmp')
+    assert res is False
+    res = call('service.start', 'snmp')
+    assert res is True
+    res = call('snmp.get_snmp_users')
+    assert "snmpSystemUser" in res
+    assert SNMP_USER_NAME in res
+
+
+def test_32_validate_SNMPv3_user_retained_across_v3_disable(request):
+    depends(request, ["SNMPv3_USER_ADD"], scope="module")
+
+    # Disable and check
+    res = call('snmp.update', {'v3': False})
+    assert SNMP_USER_NAME in res['v3_username']
+    res = call('snmp.get_snmp_users')
+    assert SNMP_USER_NAME in res
+
+    # Enable and check
+    res = call('snmp.update', {'v3': True})
+    assert SNMP_USER_NAME in res['v3_username']
+    res = call('snmp.get_snmp_users')
+    assert SNMP_USER_NAME in res
+
+
+@pytest.mark.parametrize('key,value', [
+    ('reset', ''),  # Reset systemd counters
+    ('v3_username', 'ixUser'),
+    ('v3_authtype', 'SHA'),
+    ('v3_password', 'SimplePassword'),
+    ('reset', ''),  # Reset systemd counters
+    ('v3_privproto', 'DES'),
+    ('v3_privpassphrase', 'Pass phrase with spaces'),
+    # Restore original user name
+    ('v3_username', SNMP_USER_NAME)
+])
+def test_35_validate_SNMPv3_user_changes(request, key, value):
+    """
+    Make changes to the SNMPv3 user name, password, etc. and confirm user function.
+    This also tests a pass phrase that includes spaces.
+    NOTE: We include systemd counter resets because these calls require the most restarts.
+    """
+    depends(request, ["SNMPv3_USER_ADD"], scope="session")
+    if key == 'reset':
+        # Reset the systemd restart counter
+        reset_systemd_svcs("snmpd snmp-agent")
+    else:
+        res = call('snmp.update', {key: value})
+        assert value in res[key]
+        assert get_systemctl_status('snmp-agent') == "RUNNING"
+
+        # Confirm user function after change
+        user_config = call('snmp.config')
+        res = user_list_users(user_config)
+        assert user_config['v3_username'] in res
+
+
+def test_40_validate_SNMPv3_user_delete(request):
+    depends(request, ["SNMPv3_USER_ADD"], scope="module")
```

Comment:
This the same couple of issues. Numbers in the test names that serve for nothing (tests are really executed in their definition order) and dependencies that should really be fixtures.